test/e2e/framework/framework.go

```diff
@@ -48,69 +48,91 @@ var _ ClusterInterface = &Framework{}
 func NewFramework(namePrefix string) *Framework {
 	var err error
 
-	f := &Framework{
-		name:            names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-", namePrefix)),
-		e2eArtifactsDir: "",
-		FullClient:      FullClient{},
-		clusters:        make([]*Cluster, 0, len(TestContext.RestConfigs)),
-	}
-
+	e2eArtifactsDir := ""
 	if len(TestContext.ArtifactsDir) != 0 {
-		f.e2eArtifactsDir = path.Join(TestContext.ArtifactsDir, "e2e")
-		err = os.Mkdir(f.e2eArtifactsDir, 0777)
+		e2eArtifactsDir = path.Join(TestContext.ArtifactsDir, "e2e")
+		err = os.Mkdir(e2eArtifactsDir, 0777)
 		if !os.IsExist(err) {
 			o.Expect(err).NotTo(o.HaveOccurred())
 		}
 	}
```

Comment:
Maybe we could extract code ensuring the artifact directories are set up to a helper function returning an error, and call it in `NewFramework`, asserting an error didn't occur only once?